patch_rebaser/patch_rebaser.py

```diff
@@ -261,6 +269,28 @@ def update_remote_patches_branch(self):
             self.repo.git.push(self.remote, self.tag_name)
             self.repo.git.push("-f", self.remote, self.branch)
 
+    def _rebuild_gitreview(self):
+        dlrn = get_dlrn_variables()
+        url = self.repo.repo.remote(self.remote).url
+        if isinstance(url, list):
+            url = url[0]
+        component = url.split('/')[-1]
+        host_port = url.split('/')[2]
+        host = host_port.split(':')[0]
+        port = host_port.split(':')[1]
```

Comment:
Let's split the url into a list once and do the same for the host_port once we get it. I would also add a comment describing the expected format of the strings so it's clear what we are trying to extract.